strawberryfields/engine.py

```diff
@@ -721,11 +745,27 @@ def run_async(
         if "shots" not in run_options:
             raise ValueError("Number of shots must be specified.")
 
-        return self._connection.create_job(self.target, program, run_options)
+        # Serialize a Blackbird circuit for network transmission
+        bb = to_blackbird(program)
+        bb._target["name"] = program.target
+        bb._target["options"] = run_options
+        circuit = bb.serialize()
```

Comment:
```suggestion
        bb = to_blackbird(program)
        bb._target["options"] = run_options
        circuit = bb.serialize()
```
The target name and options should be set by the `to_blackbird` function, although any potential kwargs passed to the `eng.run()` method wouldn't be in `program.run_options`, so those would still need to be updated afterwards.

Reply:
An option (to avoid setting hidden attributes) would be to create a deep copy of the program and update its `run_options` (starting from line 740):
```python
# update the run options if provided
if kwargs:
    program = copy.deepcopy(program)
    program.run_options.update(kwargs)

if "shots" not in program.run_options:
    raise ValueError("Number of shots must be specified.")

# Serialize a Blackbird circuit for network transmission
circuit = to_blackbird(program).serialize()
```
Maybe not as performant though. 🤔

Reply:
Thanks for the tips! Although I generally prefer not to access protected members, I would argue that
```python
bb._target["options"] = run_options
```
is still less sketchy than
```python
program = copy.deepcopy(program)
```
As you pointed out, the latter also has some performance implications (although these are probably negligible given the context in which the code is run).